### ISSUE TITLE
it is not handling ~ in dependencies

### ISSUE THREAD

**Issue by marcelotrevisani** [conda/grayskull#441]
Since the change you linked also contains this bug, I'll mention it explicitly: If dependencies in `pyproject.toml` use the version range shorthand (`~` or `^`), it does no conversion when populating the `meta.yaml` file. So, when presented with a dependency like so (using boto3 as an example):

```
boto3 = "~1.24"
```

it will produce
```
 - boto3 ~1.24
```

resulting in this error message
```
conda.exceptions.InvalidVersionSpec: Invalid version '~1.24': regex specs must start with '^' and end with '$'
```

---
It should instead produce the following
```
- boto3 >=1.24,<1.25
```

_Originally posted by @AlmightyYakob in https://github.com/conda-incubator/grayskull/issues/440#issuecomment-1412840767_

**Comment by AlmightyYakob** [comment]
From that same thread:

> Grayskull should be able to handle the ^ but not the ~, so, it does need some improvements, indeed.

Just for the record, the error I reported above occurs for both the `~` and `^` version range shorthands.

**Comment by xylar** [comment]
Yes, I'm seeing this, too :-(

**Comment by xylar** [comment]
This seems to be a recent regression.

**Comment by xylar** [comment]
Ah, sorry, I see that this is for `pyproject.toml`, which is not fully supported anyway.

**Comment by dlqqq** [comment]
Fixed in 2.3.0. 🎉
